File: templates/_composition/_shared/Page.Camera.AddResource/Helpers/CameraOrientationExtensions.cs
using Windows.Devices.Sensors;
using Windows.Graphics.Display;
using Windows.Storage.FileProperties;

namespace Param_RootNamespace.Helpers
{
    public static class CameraOrientationExtensions
    {
        public static SimpleOrientation ToSimpleOrientation(this DisplayInformation displayInformation, SimpleOrientation deviceOrientation, bool isFlipped)
        {
            var result = deviceOrientation;

            if (displayInformation.NativeOrientation == DisplayOrientations.Portrait)
            {
                switch (result)
                {
                    case SimpleOrientation.Rotated90DegreesCounterclockwise:
                        result = SimpleOrientation.NotRotated;
                        break;
                    case SimpleOrientation.Rotated180DegreesCounterclockwise:
                        result = SimpleOrientation.Rotated90DegreesCounterclockwise;
                        break;
                    case SimpleOrientation.Rotated270DegreesCounterclockwise:
                        result = SimpleOrientation.Rotated180DegreesCounterclockwise;
                        break;
                    case SimpleOrientation.NotRotated:
                        result = SimpleOrientation.Rotated270DegreesCounterclockwise;
                        break;
                }
            }

            if (isFlipped)
            {
                switch (result)
                {
                    case SimpleOrientation.Rotated90DegreesCounterclockwise:
                        return SimpleOrientation.Rotated270DegreesCounterclockwise;
                    case SimpleOrientation.Rotated270DegreesCounterclockwise:
                        return SimpleOrientation.Rotated90DegreesCounterclockwise;
                }
            }

            return result;
        }

        public static int ToDegrees(this DisplayOrientations orientation)
        {
            switch (orientation)
            {
                case DisplayOrientations.Portrait:
                    return 90;
                case DisplayOrientations.LandscapeFlipped:
                    return 180;
                case DisplayOrientations.PortraitFlipped:
                    return 270;
                default:
                    return 0;
            }
        }

        public static PhotoOrientation ToPhotoOrientation(this SimpleOrientation orientation, bool isFlipped)
        {
            if (isFlipped)
            {
                switch (orientation)
                {
                    case SimpleOrientation.Rotated90DegreesCounterclockwise:
                        return PhotoOrientation.Transverse;
                    case SimpleOrientation.Rotated180DegreesCounterclockwise:
                        return PhotoOrientation.FlipVertical;
                    case SimpleOrientation.Rotated270DegreesCounterclockwise:
                        return PhotoOrientation.Transpose;
                    default:
                        return PhotoOrientation.FlipHorizontal;
                }
            }

            switch (orientation)
            {
                case SimpleOrientation.Rotated90DegreesCounterclockwise:
                    return PhotoOrientation.Rotate90;
                case SimpleOrientation.Rotated180DegreesCounterclockwise:
                    return PhotoOrientation.Rotate180;
                case SimpleOrientation.Rotated270DegreesCounterclockwise:
                    return PhotoOrientation.Rotate270;
                default:
                    return PhotoOrientation.Normal;
            }
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Uwp/_comp/_shared/Page.Camera.AddResource._VB/Helpers/CameraOrientationExtensions.vb
Imports System.Runtime.CompilerServices
Imports Windows.Devices.Sensors
Imports Windows.Graphics.Display
Imports Windows.Storage.FileProperties

Namespace Helpers
    Public Module CameraOrientationExtensions

      <Extension>
      Public Function ToSimpleOrientation(displayInformation As DisplayInformation, deviceOrientation As SimpleOrientation, isFlipped As Boolean) As SimpleOrientation
          Dim result = deviceOrientation

          If displayInformation.NativeOrientation = DisplayOrientations.Portrait Then
              Select Case result
                Case SimpleOrientation.Rotated90DegreesCounterclockwise
                    result = SimpleOrientation.NotRotated
                Case SimpleOrientation.Rotated180DegreesCounterclockwise
                    result = SimpleOrientation.Rotated90DegreesCounterclockwise
                Case SimpleOrientation.Rotated270DegreesCounterclockwise
                    result = SimpleOrientation.Rotated180DegreesCounterclockwise
                Case SimpleOrientation.NotRotated
                    result = SimpleOrientation.Rotated270DegreesCounterclockwise
              End Select
          End If

          If isFlipped Then
              Select Case result
                Case SimpleOrientation.Rotated90DegreesCounterclockwise
                    Return SimpleOrientation.Rotated270DegreesCounterclockwise
                Case SimpleOrientation.Rotated270DegreesCounterclockwise
                    Return SimpleOrientation.Rotated90DegreesCounterclockwise
              End Select
          End If

          Return result
      End Function

      <Extension>
      Public Function ToDegrees(orientation As DisplayOrientations) As Integer
          Select Case orientation
              Case DisplayOrientations.Portrait
                  Return 90
              Case DisplayOrientations.LandscapeFlipped
                  Return 180
              Case DisplayOrientations.PortraitFlipped
                  Return 270
              Case Else
                  Return 0
          End Select
      End Function

      <Extension>
      Public Function ToPhotoOrientation(orientation As SimpleOrientation, isFlipped As Boolean) As PhotoOrientation
          If isFlipped Then
              Select Case orientation
                Case SimpleOrientation.Rotated90DegreesCounterclockwise
                    Return PhotoOrientation.Transverse
                Case SimpleOrientation.Rotated180DegreesCounterclockwise
                    Return PhotoOrientation.FlipVertical
                Case SimpleOrientation.Rotated270DegreesCounterclockwise
                    Return PhotoOrientation.Transpose
                Case Else
                    Return PhotoOrientation.FlipHorizontal
              End Select
          End If

          Select Case orientation
              Case SimpleOrientation.Rotated90DegreesCounterclockwise
                  Return PhotoOrientation.Rotate90
              Case SimpleOrientation.Rotated180DegreesCounterclockwise
                  Return PhotoOrientation.Rotate180
              Case SimpleOrientation.Rotated270DegreesCounterclockwise
                  Return PhotoOrientation.Rotate270
              Case Else
                  Return PhotoOrientation.Normal
          End Select
      End Function
    End Module
End Namespace
